Public api

haxys opened the pull request:
This seems stable enough for master, but we need to make sure that offload_to_backend works before we submit this one.

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Added public_api.py, created onion_info API.

Commit:
Added some comments.

artagel reviewed (commented):
We need to see if these changes roll back anything.

Commit:
Automatically add the base_url to the scan list for every link found.

Commit:
Merge branch 'master' into offload_to_backend

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Expanded fault checks.

this_onion.base_url will be set to 'Unknown' if there was a fault but no redirect.

this_onion.title will be set to 'Unknown' if we're on the base_url but there was a fault.

artagel committed:
Added setting the offline flag to false when it is offline, before we were never doing that.

artagel committed:
Updated logic to always put onions that have never been accessed first.

Commit:
Merge branch 'offload_to_backend' into public_api

Commit:
Added a 'submit_url' api call for adding new urls.

Commit:
Pruning unnecessary things.
